frontend/lib/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -167,23 +172,53 @@ function PlotlyFigure({
         // data is originalData + selectedpoints
         spec.data = data
         spec.layout.selections = selections
-
-        const hasSelectedPoints: boolean = spec.data.some(
-          (trace: any) =>
-            "selectedpoints" in trace && trace.selectedpoints.length > 0
-        )
-        if (hasSelectedPoints) {
-          spec.data.forEach((trace: any) => {
-            if (!trace.selectedpoints) {
-              // Plotly will automatically set traces in selectedpoints with the 100% opaqueness
-              // setting selectedpoints to an empty array will set these points to be opaque, thus representing that they're unselected
-              trace.selectedpoints = []
-            }
-          })
+        if (spec.data) {
+          const hasSelectedPoints: boolean = spec.data.some(
+            (trace: any) =>
+              "selectedpoints" in trace &&
+              trace.selectedpoints &&
+              trace.selectedpoints.length > 0
+          )
+          if (hasSelectedPoints) {
+            spec.data.forEach((trace: any) => {
+              if (!trace.selectedpoints) {
+                // Plotly will automatically set traces in selectedpoints with the 100% opaqueness
+                // setting selectedpoints to an empty array will set these points to be opaque, thus representing that they're unselected
+                trace.selectedpoints = []
+              }
+            })
+          }
         }
       }
     }
-
+    const rangeEvent = widgetMgr.getElementState(element.id, RANGE)
+    const autoRangeEvent = widgetMgr.getElementState(element.id, AUTORANGE)
+    const dragmode = widgetMgr.getElementState(element.id, DRAGMODE)
+
+    try {
+      if (rangeEvent && rangeEvent["xaxis.range[0]"]) {
+        spec.layout.xaxis.range = [
+          rangeEvent["xaxis.range[0]"],
+          rangeEvent["xaxis.range[1]"],
+        ]
+        spec.layout.yaxis.range = [
+          rangeEvent["yaxis.range[0]"],
+          rangeEvent["yaxis.range[1]"],
+        ]
+      } else if (autoRangeEvent && autoRangeEvent["xaxis.autorange"]) {
+        spec.layout.xaxis.autorange = true
+        spec.layout.yaxis.autorange = true
+      }
+      if (dragmode) {
+        spec.layout.dragmode = dragmode
+      }
+    } catch (e) {
+      logMessage(e)
+      logMessage(spec)
+      logMessage(rangeEvent)
+      logMessage(autoRangeEvent)
+      logMessage(dragmode)
```

Comment:
Will this be removed once we merge this? Or do you want to keep that in>

Reply:
I was hoping to keep that in so that when this catches, we can ask for logs.